File: library/src/main/java/com/zone/zbanner/indicator/animation/DefaultAnimation.java
package com.zone.zbanner.indicator.animation;

import com.zone.zbanner.indicator.IndicatorView;
import com.zone.zbanner.indicator.animation.abstarct.BaseAnimation;

/**
 * Created by Zone on 2016/1/28.
 */
public class DefaultAnimation extends BaseAnimation {


    public DefaultAnimation(IndicatorView indicatorView, int itemLength) {
        super(indicatorView, itemLength);
    }
    @Override
    public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
//        String s = String.format("onPageScrolled====position:%d /tpositionOffset:%f /tpositionOffsetPixels:%d /t", position, positionOffset, positionOffsetPixels);
//        System.out.println(s);
    }

    @Override
    public void onPageSelected(int position) {
        super.onPageSelected(position);
        indicatorView.getIv_Top().setX( itemLength * position);
    }
}

File: library/src/main/java/com/zone/zbanner/indicator/animation/DefaultAnimation.kt
package com.zone.zbanner.indicator.animation

import com.zone.zbanner.indicator.IndicatorView
import com.zone.zbanner.indicator.animation.abstarct.BaseAnimation

/**
 * Created by Zone on 2016/1/28.
 */
class DefaultAnimation(indicatorView: IndicatorView, itemLength: Int) : BaseAnimation(indicatorView, itemLength) {
    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}

    override fun onPageSelected(position: Int) {
        super.onPageSelected(position)
        indicatorView.ivTop.x = (itemLength * position).toFloat()
    }
}
